lib/charms/observability_libs/v0/kubernetes_service_patch.py

```diff
@@ -197,7 +218,9 @@ def _patch(self, _) -> None:
 
         client = Client()
         try:
-            client.patch(Service, self._app, self.service, patch_type=PatchType.MERGE)
+            if self.service_name != self._app:
+                self._delete_and_create_service(client)
```

Comment:
It looks like you're missing an else here? Presumably no need to patch the service you've just created?

Reply:
Well it wasn't really a mistake. The service we create during `_delete_and_create_service` doesn't have the ports/labels that the user provided. It's simply the same old service with a new name.